meerk40t/core/spoolers.py

```diff
@@ -812,7 +815,11 @@ def remove(self, element, index=None):
                     )
                 except AttributeError:
                     pass
-                self._queue.remove(element)
+                try:
+                    self._queue.remove(element)
+                except ValueError:
+                    # We might have waited for too long, the job is no longer there...
```

Comment:
with self._lock:

Reply:
We have a thread.Lock() value that we used above. If for some reason the job could have been removed elsewhere while we're doing the work that elsewhere needs to be also put into the synchronization gate. Whatever changes the queue should be locked so only one thing fiddles with that at a time.

Reply:
Its a simple case to test, where a lock makes limited sense: run job, right click in spoolerpanel while it still runs, select but don't execute yet the option to cancel the job, job terminates while context menu is still open, click...
A lock would not help.